src/react-components/change-scene-dialog.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styles from "../assets/stylesheets/change-scene-dialog.scss";
+import cx from "classnames";
+import DialogContainer from "./dialog-container.js";
+import { handleTextFieldFocus, handleTextFieldBlur } from "../utils/focus-utils";
+import { WithHoverSound } from "./wrap-with-audio";
+
+export default class ChangeSceneDialog extends Component {
+  state = {
+    url: ""
+  };
+
+  static propTypes = {
+    onChange: PropTypes.func,
+    onClose: PropTypes.func
+  };
+
+  onUrlChange = e => {
+    this.setState({
+      url: e.target && e.target.value
+    });
+  };
+
+  reset = e => {
+    e.preventDefault();
+    this.setState({ url: "" });
+  };
+
+  onChangeClicked = e => {
+    e.preventDefault();
+
+    if (this.state.url) {
+      this.props.onChange(this.state.url);
+    }
+
+    this.props.onClose();
+  };
+
+  render() {
+    const { onChange, onClose, ...other } = this.props; // eslint-disable-line no-unused-vars
```

Comment:
I dunno how I feel about this catch-all props pattern. I'd rather be explicit where we can, as is the case with this dialog. For one thing, we probably shouldn't encourage use of `eslint-disable-line`.

Reply:
ah this was copied from elsewhere. i'll remove it though